lib/segment/src/id_tracker/simple_id_tracker.rs

```diff
@@ -240,6 +240,13 @@ impl IdTracker for SimpleIdTracker {
         Ok(())
     }
 
+    fn drop_by_internal_id(&mut self, internal_id: PointOffsetType) -> OperationResult<()> {
+        match self.internal_to_external.get(internal_id as usize) {
```

Comment:
`internal_to_external` is a vector and `get` checks only the vector's range. If point is deleted, `internal_to_external[internal_id]` contains placeholder value `u64::MAX`. Instead of calling `self.internal_to_external.get` we can use just `Self::external_id` method.